src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -296,6 +299,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         notebook: NotebookDocument;
         controller: VSCodeNotebookController;
     }) {
+        this.widgetCoordinator.setActiveController(event.notebook, event.controller);
```

Comment:
I think this handleOnNotebookControllerSelected is the event that we're currently sometimes not getting on first document open, hopefully that's fixes now in notebook/dev

Reply:
Yes, i've run into that and filed an issue in VSC.
I just change the kernel to get things working.